src/D2L.CodeStyle.Analyzers/Immutability/StatelessFuncAnalyzer.cs

```diff
@@ -77,6 +77,11 @@ ImmutableHashSet<ISymbol> statelessFuncs
 				return;
 			}
 
+			// static functions can't capture state
+			if ( argument.IsStaticFunction() ) {
+				return;
+			}

```

Comment:
Pulled out common code to an extension and lifted this outside the switch

Reply:
nit: collapse and change the pattern in the switch to `AnonymousFunctionExpressionSyntax` and move this there? Diagnostic message can change slightly to accommodate.

It used to be more collapsed, but I guess I missed the base class when switching to static funcs, https://github.com/Brightspace/D2L.CodeStyle/commit/a826fd3e49052dfa9314307454147ee238855a4d#diff-5b07894d106bc2ac1710cd24766f935abaa2a2076f85dc35925a2336c7e2cb96

Reply:
Ah, I thought they were totally different diagnostics but you're right it just varies by the message.
I can do that but I'll do it as a follow-up (since it's a totally different analyzer)